crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -697,19 +703,24 @@ where
 
             // we need to first check the buffer for the head and its ancestors
             let lowest_unknown_hash = self.lowest_buffered_ancestor_or(target);
-
-            trace!(target: "consensus::engine", request=?lowest_unknown_hash, "Triggering pipeline with target instead of downloading");
-
-            self.sync.set_pipeline_sync_target(lowest_unknown_hash);
+            trace!(target: "consensus::engine", request=?lowest_unknown_hash, "Triggering full block download for missing ancestors of the new head");
+            lowest_unknown_hash
         } else {
             // we need to first check the buffer for the head and its ancestors
             let lowest_unknown_hash = self.lowest_buffered_ancestor_or(state.head_block_hash);
-
             trace!(target: "consensus::engine", request=?lowest_unknown_hash, "Triggering full block download for missing ancestors of the new head");
+            lowest_unknown_hash
+        };
 
+        // if the threshold is zero, we should not download the block first, and just use the
+        // pipeline. Otherwise we use the tree to insert the block first
+        if self.pipeline_run_threshold == 0 {
```

Comment:
should we even support this edge case?

Reply:
I included this so we could force the pipeline to run in tests that require `TestStages` to run on forkchoice updates, for example `unknown_finalized_hash`. Otherwise we will try to download a block when a test sends a FCU, which will attempt to download a block and call the `NoopFullBlockClient`. This returns nothing so the full block futures will continue to be polled and these tests will stall.

See usages of `disable_blockchain_tree_sync` for all tests that stall without this check.

Reply:
this is reasonable